obstore/python/obstore/fsspec.py

```diff
@@ -478,25 +490,71 @@ def __init__(
         **kwargs: Any,
     ) -> None:
         """Create new buffered file."""
-        if mode != "rb":
-            raise ValueError("Only 'rb' mode is currently supported")
         super().__init__(fs, path, mode, **kwargs)
 
-    def read(self, length: int = -1) -> Any:
+        bucket, self.path = self.fs.split_path(path)
+        self.store = self.fs.construct_store(bucket)
+
+        self.mode = mode
+
+        if self.mode == "rb":
+            self._reader = open_reader(self.store, self.path)
+
+    def read(self, length: int = -1) -> Bytes:
         """Return bytes from the remote file.
 
         Args:
             length: if positive, returns up to this many bytes; if negative, return all
                 remaining bytes.
 
+        Returns:
+            Data in bytes
+
         """
         if length < 0:
-            data = self.fs.cat_file(self.path, self.loc, self.size)
-            self.loc = self.size
-        else:
-            data = self.fs.cat_file(self.path, self.loc, self.loc + length)
-            self.loc += length
-        return data
+            length = self.size - self.loc
+
+        self._reader.seek(self.loc)
```

Comment:
We shouldn't seek on every underlying read call if we don't need to. Seeking discards the internal buffer kept by rust

(Forgot to post this comment earlier)